src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -286,6 +288,20 @@ public ChainedHeaderBlock GetChainedHeaderBlock(uint256 blockHash)
             return chainedHeaderBlock;
         }
 
+        public ChainedHeaderBlock[] GetChainedHeaderBlocks(List<uint256> blockHashes)
+        {
+            var chainedHeaderBlocks = blockHashes.ToDictionary(x => x, x => (ChainedHeaderBlock)null);
```

Comment:
Creating a dictionary and intersecting the hashes below feels a bit overkill. It could also be expensive, cause `this.chainedHeadersByHash` can have thousands of entries in it. Why not just check that the hash is in `this.chainedHeadersByHash`?

Reply:
I ran a test with ~90,000 entries in each - of which ~40,000 intersect. Without the Intersect it takes 2.5 times as long..